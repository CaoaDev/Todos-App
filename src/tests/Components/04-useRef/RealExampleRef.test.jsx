import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../Components/04-useRef/RealExampleRef';
import { MultipleCustomHook } from '../../../Components/03-examples/MultipleCustomHook';


describe('Pruebas en <RealExampleRef />', () => {
    test('Debe mostrarse conrrectamente', () => {
        const wrapper= shallow( <RealExampleRef /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find ( 'MultipleCustomHook' ).exists() ).toBeFalsy;
    });

    test('Debe mostrar el componente <MultipleCustomHook />', () => {
        const wrapper= shallow( <MultipleCustomHook /> );
        wrapper.find( 'button' ).simulate( 'click' );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find ( 'MultipleCustomHook' ).exists() ).toBeFalsy;
    });
})