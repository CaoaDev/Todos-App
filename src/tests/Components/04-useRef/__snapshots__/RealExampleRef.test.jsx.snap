// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> Debe mostrar el componente <MultipleCustomHook /> 1`] = `
<div>
  <h1>
    <center>
       BreakingBad Quotes!!! 
    </center>
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Frase...
  </button>
</div>
`;

exports[`Pruebas en <RealExampleRef /> Debe mostrarse conrrectamente 1`] = `
<div>
  <h2>
     RealExampleRef 
  </h2>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
